pkg/api/component_report.go

```diff
@@ -234,9 +268,9 @@ func (c *componentReportGenerator) getJobRunTestStatusFromBigQuery() (
 						COUNT(*) AS total_count,
```

Comment:
I wonder if replacing _COUNT(*) AS total_count_ with _COUNT(distinct(file_path)) AS total_count_ will work.

Reply:
I think that'll count the total number of tests for a particular job, including the  multiple failures

Reply:
I tried this (_with COUNT(distinct(file_path)) AS total_count_ replacing _COUNT(cm.id) AS total_count_), it seems to work as expected (14 total vs 16 total):

WITH latest_component_mapping AS (
						SELECT *
						FROM ci_analysis_us.component_mapping cm
						WHERE created_at = (
								SELECT MAX(created_at)
								FROM openshift-gce-devel.ci_analysis_us.component_mapping))
					SELECT
						ANY_VALUE(test_name) AS test_name,
						ANY_VALUE(testsuite) AS test_suite,
						cm.id as test_id,
						COUNT(distinct(file_path)) AS total_count,
						SUM(success_val) AS success_count,
						SUM(flake_count) AS flake_count,
						network,
						upgrade,
						arch,
						platform,
						flat_variants,
						ANY_VALUE(variants) AS variants,
						ANY_VALUE(cm.component) AS component,
						ANY_VALUE(cm.capabilities) AS capabilities
					FROM ci_analysis_us.junit
					INNER JOIN latest_component_mapping cm ON testsuite = cm.suite AND test_name = cm.name
					WHERE
						TIMESTAMP(modified_time) >= "2023-06-21T00:00:00Z" AND TIMESTAMP(modified_time) < "2023-06-28T23:59:59Z"
						AND (prowjob_name LIKE 'periodic-%%' OR prowjob_name LIKE 'release-%%' OR prowjob_name LIKE 'aggregator-%%')  AND test_name NOT LIKE '%disruption/%' AND branch = "4.14" AND
            test_name="[sig-instrumentation] Prometheus [apigroup:image.openshift.io] when installed on the cluster shouldn't report any alerts in firing state apart from Watchdog and AlertmanagerReceiversNotConfigured [Early][apigroup:config.openshift.io] [Skipped:Disconnected] [Suite:openshift/conformance/parallel]" AND
            network="ovn" AND upgrade="no-upgrade" AND arch="amd64" AND platform="vsphere-upi"
					GROUP BY
						network,
						upgrade,
						arch,
						platform,
						flat_variants,
						cm.id 
          ORDER BY cm.id, network, upgrade, arch, platform, flat_variants

Reply:
That does seem to work but I'm struggling to understand why.  file_path is just the gcs bucket path, right? Why does distinctly counting those result in the right count? Maybe it'll be clearer to me with fresh eyes

Reply:
file_path is unique per job run. We can probably use prowjob_build_id if that is also unique.

Reply:
The build is is the right unique identifier for a job run, if you're counting file_path then you could have multiple per job when there's multiple junit.xml.  Not sure which is the right to count.

This is effectively counting the number of unique job runs as a proxy for the number of unique test results. It works here but is there any situation where it doesn't work?

Reply:
I think this works to count file_path, very clever.. thanks!